src/OpenCvSharp/Modules/core/Struct/RotatedRect.cs

```diff
@@ -39,6 +39,47 @@ public RotatedRect(Point2f center, Size2f size, float angle)
         Angle = angle;
     }
 
+    /// <summary>
+    /// Any 3 end points of the RotatedRect. They must be given in order (either clockwise or anticlockwise).
+    /// </summary>
+    public RotatedRect(Point2f point1, Point2f point2, Point2f point3)
+    { // https://github.com/opencv/opencv/blob/6ad77b23193bdf7e40db83e6077789284ac08781/modules/core/src/types.cpp#LL147C20-L147C20
+        var center = (point1 + point3) * 0.5;
+        var vecs = new Vec2f[2];
+        vecs[0] = (point1 - point2).ToVec2f();
+        vecs[1] = (point2 - point3).ToVec2f();
+        var x = Math.Max(Cv2.Norm(point1.ToVec2f()), Math.Max(Cv2.Norm(point2.ToVec2f()), Cv2.Norm(point3.ToVec2f())));
+        var a = Math.Min(Cv2.Norm(vecs[0]), Cv2.Norm(vecs[1]));
+
+        const float fltEpsilon = 1.19209290e-7f; // https://github.com/opencv/opencv/blob/6ad77b23193bdf7e40db83e6077789284ac08781/modules/dnn/src/math_utils.hpp#L39
+        static double Ddot(Vec2f a, Vec2f b)
+        { // https://github.com/opencv/opencv/blob/0052d46b8e33c7bfe0e1450e4bff28b88f455570/modules/core/include/opencv2/core/matx.hpp#L741
+            var s = 0d;
+            for (var i = 0; i < 2; i++)
+            {
+                s += (double)a[i] * b[i];
+            }
+            return s;
+        }
+        // check that given sides are perpendicular
+        Debug.Assert(Math.Abs(Ddot(vecs[0], vecs[1])) * a <= fltEpsilon * 9 * x * (Cv2.Norm(vecs[0]) * Cv2.Norm(vecs[1])));
```

Comment:
This assert will fail when providing following points: `new RotatedRect(new(403.2502, 109.6314), new(505.6677, 349.0594), new(271.8243, 446.3848))`

Reply:
> The name 'Debug' does not exist in the current context

How do we do asserts in methods translated from cpp?
IIRC, OpenCV assert will always throw runtime cpp exception.

Reply:
```c#
using System.Diagnostics; // Add this!!

...

Debug.Assert(...);
```

Reply:
May we use [`global using`](https://blog.jetbrains.com/dotnet/2021/11/18/global-usings-in-csharp-10/)?